test/autorest/generated/lrogroup/models.go

```diff
@@ -0,0 +1,851 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package lrogroup
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"net/http"
+)
+
+type CloudError struct {
+	Message *string `json:"message,omitempty"`
+	Status  *int32  `json:"status,omitempty"`
+}
+
+func newCloudError(resp *azcore.Response) error {
+	err := CloudError{}
+	if err := resp.UnmarshalAsJSON(&err); err != nil {
+		return err
+	}
+	return err
+}
+
+func (e CloudError) Error() string {
+	msg := ""
+	if e.Message != nil {
+		msg += fmt.Sprintf("Message: %v\n", *e.Message)
+	}
+	if e.Status != nil {
+		msg += fmt.Sprintf("Status: %v\n", *e.Status)
+	}
+	if msg == "" {
+		msg = "missing error info"
+	}
+	return msg
+}
+
+// LrOSCustomHeaderPost202Retry200Options contains the optional parameters for the LrOSCustomHeader.Post202Retry200 method.
+type LrOSCustomHeaderPost202Retry200Options struct {
+	// Product to put
+	Product *Product
+}
+
+// LrOSCustomHeaderPost202Retry200Response contains the response from method LrOSCustomHeader.Post202Retry200.
+type LrOSCustomHeaderPost202Retry200Response struct {
+	// Location contains the information returned from the Location header response.
+	Location *string
```

Comment:
Our header codegen is working great, but unfortunately this is introducing a bunch of response envelopes and thus pollers which seems unfortunate.

The following headers are implementation details for LROs.

- Azure-AsyncOperation
- Location
- Retry-After

Some swaggers include them however it's not necessary as the ARM LRO engine knows about them.  I haven't thought too deeply about this but I'm thinking that we should omit these headers from codegen which should remove all of these response envelopes and pollers.  Thoughts?

Reply:
Well, as you said, I think there could be other services that will use some/all of them. I am worried about removing them in case some service uses them and there is value in exposing them instead of hiding them away from users for cases that are not LRO.

Reply:
They'd use them as an implementation detail to polling, I don't think exposing them like this has any value but I could be wrong (it could be the case we do this for ARM but not data-plane).  We should discuss this in scrum.

Reply:
Hmmm well if you think that this is the only case they would be used then we should remove them. I'll add an issue.

Reply:
Here's the issue #358